src/main/java/org/opendatakit/aggregate/client/popups/PublishPopup.java

```diff
@@ -294,4 +295,8 @@ public native static boolean validateEmail(String email) /*-{
     return re.test(String(email).toLowerCase());
   }-*/;
 
+  public boolean validateURL(String url){
+    return url.matches("(http?|https?)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]");
```

Comment:
This feels fragile to me. I think we either do something very naive (just check for `(http?|https?)://`) or we use the internal Java library (or something from Apache if it's already embedded).

Reply:
I thought about using `URL` for that, but we can't use it because there's no GWT translation of `URL`. 

Unfortunately, there's no built-in URL validation in JavaScript, so people use regexps like the one @sd1998 is proposing (there are plenty of custom validators out there).

The js validation library [validate.js](https://validatejs.org/#validators-url) uses [this regexp to validate URLs](https://gist.github.com/dperini/729294). we could use that one instead.